### PR TITLE
Enable the teacher feedback tab for elementary-level courses (TEACH-634)

### PR DESCRIPTION
**Pull request by etaderhold**
This change is to enable the Teacher Feedback interface for elementary-level courses where it is currently disabled.

Teacher view screenshot:
image

Student view screenshot:
image

## Links

[Spec](https://docs.google.com/document/d/1WA5cEFoaAZRXi1nbVrHmTKAGtf5q88FU3w_y0MfQJCM/edit#heading=h.35irlwfouroh)
[JIRA](https://codedotorg.atlassian.net/browse/TEACH-634)

## Testing story

Unit test added.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by hannahbergam — APPROVED** [review]
Seems simple enough, and thanks for adding that test. LGTM!

**Comment by dju90** [comment]
Is there any follow-up work we need to do here to clean up the previous logic?

**Comment by etaderhold** [comment]
> Is there any follow-up work we need to do here to clean up the previous logic?

What sort of clean up are you thinking of? The `isCSDorCSP` variable I'm removing from the display logic here is still used elsewhere in this component, so that doesn't need to be removed. Hannah mentioned I might need to approve some image test differences when the dev of the day encounters those. Nothing else seems immediately obvious as follow-up work. I certainly could be missing something though!

**Comment by dju90** [comment]
Sounds good! Just checking :)

**Comment by dju90** [comment]
One quick note - if you can hold off on merging this until I check in with our professional learning team, that would be great. That shouldn't take long.

**Comment by bethanyaconnor** [comment]
Do you mind checking that this behavior looks expected on an hour of code unit, such as dance (localhost-studio.code.org:3000/s/dance-2019)? We have some extra caching logic for hour of code units that can sometimes cause funkiness.

**Comment by etaderhold** [comment]
> Do you mind checking that this behavior looks expected on an hour of code unit, such as dance (localhost-studio.code.org:3000/s/dance-2019)? We have some extra caching logic for hour of code units that can sometimes cause funkiness.

Thanks for the suggestion! The feedback tab shows up on an Hour of Code unit, but the API request to set the feedback (`POST /api/v1/teacher_feedbacks`) fails with a 422 error. Looks like a CSRF token is not being passed in this case.

Possible the lack of such a token is part of the caching and performance optimizations you mentioned here, Bethany. Bryan: is it then best to continue suppressing the Feedback tab for Hour of Code, or do we need to find a way around this?

**Comment by dju90** [comment]
I'm guessing that finding a way around the 422 is costlier/has more uncertainty in the work involved than suppressing for HoC? If so, then let's continue suppressing for HoC

**Comment by etaderhold** [comment]
> I'm guessing that finding a way around the 422 is costlier/has more uncertainty in the work involved than suppressing for HoC?

Definitely looks that way, yes.

> If so, then let's continue suppressing for HoC

🖖

**Review by bethanyaconnor — APPROVED** [review]

**Comment by etaderhold** [comment]
Reviving this PR now that Bryan has confirmed that we should enable feedback. Added requirement is that the elementary-level courses should not require verified teacher status to leave feedback. That's incorporated in here now too. Please take another look if you like!

File: apps/src/templates/instructions/teacherFeedback/EditableTeacherFeedback.jsx
Comment (bethanyaconnor) [inline apps/src/templates/instructions/teacherFeedback/EditableTeacherFeedback.jsx:241]: Should this message be changed? It currently says "You must be a verified teacher to leave feedback."

**Review by bethanyaconnor — APPROVED** [review]
One text change suggestion but otherwise looks good!

File: apps/src/templates/instructions/teacherFeedback/EditableTeacherFeedback.jsx
Comment (etaderhold) [reply apps/src/templates/instructions/teacherFeedback/EditableTeacherFeedback.jsx:241]: I believe this message is still good. We're still requiring verification for feedback in the upper-level courses. `allowEditing` is set to `verifiedInstructor || allowUnverified`, so if it's false it means the teacher is unverified and in a course that requires verification to leave feedback.

Comment (bethanyaconnor) [reply apps/src/templates/instructions/teacherFeedback/EditableTeacherFeedback.jsx:241]: oh right! sounds good

**Pull request merged**
